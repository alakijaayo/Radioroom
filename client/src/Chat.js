import React, { Component } from 'react';
import posed, { PoseGroup } from 'react-pose';

const Item = posed.li({});

class Chat extends Component {
  constructor(props) {
    super();
    this.state = { message: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ message: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.addChatMessage(this.state.message);
    this.setState({ message: '' });
  }

  render() {
    return (
      <div className="messages">
        <h2>Chat</h2>
        <form onSubmit={this.handleSubmit}>
          <textarea
            rows="5"
            cols="50"
            value={this.state.message}
            onChange={this.handleChange}
          />
          <input type="submit" value="Submit" />
        </form>
        <div className="messagelist">
          {this.props.messages ? (
            <ul>
              <PoseGroup>
                {this.props.messages.map(message => (
                  <Item key={message.id}>
                    <div id={message.id}>
                      <p>{message.text}</p>
                      <img src={message.user_url} alt="user profile" />
                    </div>
                  </Item>
                ))}
              </PoseGroup>
            </ul>
          ) : (
            ''
          )}
        </div>
      </div>
    );
  }
}

export default Chat;
