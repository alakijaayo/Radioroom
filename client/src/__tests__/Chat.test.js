import React from 'react';
import ReactDom from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Chat from '../Chat'

Enzyme.configure({ adapter: new Adapter() });

const chat = [{
  id: 1,
  text: 'hello radioroom',
  user_url: 'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/1381887_10201346045386534_973307961_n.jpg?_nc_cat=110&_nc_ht=scontent.xx&oh=469ec1722c45310e5fd3e5043e7eed66&oe=5C6DE405'
}, {

  id: 2,
  text: 'goodbye radioroom',
  user_url: 'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/15978093_10154712760570568_4766216528086979_n.jpg?_nc_cat=106&_nc_ht=scontent.xx&oh=e2864c83a2c5173327b5b548bee2a8be&oe=5C69475A'
}]

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Chat messages={chat} />, div);
  ReactDom.unmountComponentAtNode(div);
});

describe('Chat component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Chat />).toJSON();
    expect(tree).toMatchSnapshot()
  });

  // it('renders the chat messages', () => {
  //   const chatUI = mount(<Chat messages={chat}/>);
  //   expect(
  //     chatUI.containsMatchingElement(
  //       <p>hello radioroom</p>
  //     ).to.equal(true)
  //   )
  // });
});
