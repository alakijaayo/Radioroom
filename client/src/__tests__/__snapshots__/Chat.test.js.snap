// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component matches the snapshot 1`] = `
<div
  className="messages"
>
  <h2>
    Chat
  </h2>
  <div
    className="mx-auto"
    style={
      Object {
        "maxWidth": 800,
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <textarea
          className="form-control"
          cols="25"
          onChange={[Function]}
          rows="3"
          value=""
        />
        <div
          className="input-group-append"
        >
          <input
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>
  <div
    className="messagelist list-group mx-auto"
    style={
      Object {
        "maxWidth": 800,
      }
    }
  >
    
  </div>
</div>
`;
